lib/logstash/inputs/udp.rb

```diff
@@ -4,6 +4,7 @@
 require "logstash/namespace"
 require "socket"
 require "stud/interval"
+require "ipaddr"
```

Comment:
I suggest removing any other required of ipaddr like https://github.com/logstash-plugins/logstash-input-udp/blob/master/spec/support/client.rb#L3

Reply:
meh - if you insist I canl do it but OTOH it's totally ok to require a lib that is used in the file. For example, if `udp.rb` stops using it but it's still in use in `client.rb` then it won't break. In fact I'd argue that this is preferable actually.

Reply:
actually ... doing that risks reproducing the exact same problem we are solving here :P

Reply:
I guess my argument comes from ipaddr now being a runtime dependency in the gemspec so it *must* be required in the "lib/" files not the "spec"

That said I'm not blocking this PR b/c of this, up to you.
LGTM

Reply:
« ipaddr now being a runtime dependency in the gemspec so it must be required in the "lib/" files not the "spec"» not following what you mean!?

Reply:
>  ipaddr now being a runtime dependency in the gemspec

I forgot ipaddr is stdlib, but my argument was that any runtime dependency should be included in the `lib/logstash/inputs/udp.rb`, which means we'd never need to include in the specs.

I'm still LGTM on this.

Reply:
I personally think that not having an explicit `require "somelib"` in a file that explicitly uses `Someclass` is an anti-pattern.  In this case the `IPAddr` class in `client.rb` is used independently for its usage in `udp.rb` and if one day `IPAddr` is not used anymore in `udp.rb` for whatever reason, that does not mean that `IPAddr` will not be used in `client.rb` and following your suggestion of removing `require "ipaddr"` from `client.rb` it would result in a bug when running specs.

Reply:
Makes sense, agreed.